### DISCUSSION TITLE
[share] Detects if the current item has duplicate items

### DISCUSSION THREAD

**Discussion by northword** [windingwind/zotero-actions-tags#133] [Action Scripts]
## Description

Automatically detects if the current item has duplicate items.

This is an `Action` implementation of the [Linter for Zotero](https://github.com/northword/zotero-format-metadata) plugin's duplicate item detection feature.

## Action Settings

Event: Create Item
Type: Script
Data:
```js
// Automatically detects if the current item has duplicate items.
// @author northword
// @link https://github.com/windingwind/zotero-actions-tags/discussions/133
// @see https://github.com/northword/zotero-format-metadata/blob/main/src/modules/rules/item-no-duplicate.ts

const TIPWORD = "当前条目存在重复条目，请前往合并";

if (!item) {
	return;
}

const itemID = item.id;
const duplicates = new Zotero.Duplicates("1");
const search = (await duplicates.getSearchObject());
const searchResult = await search.search();
if (searchResult.includes(itemID)) {
	return TIPWORD;
} else {
	return;
}
```

**Comment by MarekAdam** [comment]
Can somebody add script option to delete detected duplicate?

**Reply by northword** [reply]
Use zoplicate plugin.

**Comment by LumaFilter** [comment]
感谢大佬，顺便小小拓展一下，检测到重复自动添加一个标签：
[问描述和实现详见这里](https://github.com/windingwind/zotero-actions-tags/discussions/310)
